src/sage/tensor/modules/free_module_automorphism.py

```diff
@@ -1085,13 +1135,62 @@ def det(self):
             1
 
         """
-        self.matrix() # forces the update of the matrix in the module's default
-                      # basis, to make sure that the dictionary self._matrices
-                      # is not empty
-        return next(iter(self._matrices.values())).det() # pick a random value in the
-                                                # dictionary self._matrices
-                                                # and compute the determinant
+        return self._some_matrix().det
+
+    determinant = det
+
+    @lazy_attribute
+    def fcp(self):
```

Comment:
Same question as for `characteristic_polynomial`:
```suggestion
    def fcp(self, var):
```